File: 0075-sort-colors/0075-sort-colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        l = 0
        zero = 0
        end = len(nums) - 1
        
        while l <= end:
            if nums[l] == 0:
                nums[l], nums[zero] = nums[zero], nums[l]
                l += 1
                zero += 1
            elif nums[l] == 1:
                l += 1
            else:
                nums[l], nums[end] = nums[end], nums[l]
                end -= 1
        

File: 0075-sort-colors/0075-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int l = 0;
        int r = 0;
        int end = nums.size() - 1;
        while(l <= end){
            if(nums[l] == 0){
                int temp = nums[l];
                nums[l] = nums[r];
                nums[r] = temp;
                l++;
                r++;
            }else if(nums[l] == 1){
                l++;
            }else{
                int temp = nums[l];
                nums[l] = nums[end];
                nums[end] = temp;
                end--;
            }
        }
        
    }
};